src/components/ListView/ListViewDataSource.js

```diff
@@ -0,0 +1,194 @@
+import invariant from 'fbjs/lib/invariant'
+
+export default class ListViewDataSource {
```

Comment:
That particular class was build using flow types incorporated into the code. I (for some reason..) thought that was not supported by the react-native-web toolchain (I had done some spot-checks and could not find any flow enabled code), so instead made a port of it (the code is nearly identical though).
I've just checked and the original code compiles just fine and works as expected. I agree we should use that. I only had to change the require's as the top of the file to use:

```js
var invariant = require('fbjs/lib/invariant');
var isEmpty = require('fbjs/lib/isEmpty');
var warning = require('fbjs/lib/warning');
```

Furthermore, the class throws a ton of linting errors because semi-colons are not allowed. I think it would be nice if we could drop code from 'react-native' in without having to remove all semi-colons (or change the require's).